app/controllers/inclusion_connect_controller.rb

```diff
@@ -0,0 +1,53 @@
+class InclusionConnectController < ApplicationController
+  skip_before_action :authenticate_agent!
+
+  def auth
+    state = set_session_state_for_inclusion_connect
+    redirect_to InclusionConnect.auth_path(state, inclusion_connect_callback_url), allow_other_host: true
+  end
+
+  def callback
+    unless params[:state] == session[:ic_state]
+      handle_failed_authentication
+      return
+    end
+
+    agent = InclusionConnect.agent(params[:code], inclusion_connect_callback_url)
+
+    if agent
+      handle_successful_authentication(agent)
+    else
+      Sentry.capture_message("Failed to authenticate agent with InclusionConnect")
+      handle_failed_authentication
+    end
+  end
+
+  private
+
+  def set_session_state_for_inclusion_connect
+    session[:ic_state] = Digest::SHA1.hexdigest("InclusionConnect - #{SecureRandom.hex(13)}")
+  end
+
+  def handle_successful_authentication(agent)
+    @current_agent = agent
+    set_session_credentials_for_inclusion_connect
+    session[:connected_with_inclusionconnect] = true
+    redirect_to root_path
```

Comment:
```suggestion
    redirect_to session.delete(:agent_return_to) || root_path
```
Comme dans le `sessions_controller`